nvtabular/tools/dataset_inspector.py

```diff
@@ -0,0 +1,191 @@
+#
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import json
+
+import cudf
+import fsspec
+import numpy as np
+from dask.distributed import Client
+from dask_cuda import LocalCUDACluster
+
+from nvtabular.io import Dataset
+
+
+# Class to help Json to serialize the data
+class NpEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, np.integer):
+            return int(obj)
+        elif isinstance(obj, np.floating):
+            return float(obj)
+        elif isinstance(obj, np.bool_):
+            return bool(obj)
+        elif isinstance(obj, np.ndarray):
+            return obj.tolist()
+        else:
+            return super(NpEncoder, self).default(obj)
+
+
+class DatasetInspector:
+    """
+    Analyzes an existing dataset to extract its statistics.
+    """
+
+    def __get_stats(self, ddf, ddf_dtypes, col, data, col_type, key_names):
+        """
+        Parameters
+        -----------
+        ddf : dask.dataframe.DataFrame
+            Dask dataframe with the data
+        ddf : dask.dataframe.DataFrame
+            Dask dataframe with the correct dtypes
+        col: string

```

Comment:
Can we use the numpy docstring guide here for https://numpydoc.readthedocs.io/en/latest/format.html#method-docstrings (see parameters).
```suggestion
        col: str
```

Reply:
Sorry about that, I will use the guide.